listenbrainz/db/stats.py

```diff
@@ -100,9 +100,10 @@ def insert_multiple_user_jsonb_data(data):
                   , {from_ts}
                   , {to_ts}
                   , NOW()
-               FROM (VALUES %s) AS t(user_name, count, stats)
-               JOIN "user"
-                 ON "user".musicbrainz_id = user_name
+               FROM (VALUES %s) AS t(user_id, count, stats)
+               -- this JOIN serves no other purpose than to filter out users for whom stats were calculated but
+               -- no longer exist in LB. if we don't filter, we'll get a FK conflict when such a case occurs
+               JOIN "user" ON "user".id = user_id 
```

Comment:
I guess this could be rewritten to use a CTE and WHERE but not sure if that's more readable?

Reply:
fine as is.